erlang/api_cowboy/apps/api_cowboy/src/api_cowboy_handler.erl

```diff
@@ -0,0 +1,145 @@
+-module(api_cowboy_handler).
+-behaviour(cowboy_handler).
+
+-export([init/2]).
+
+init(Req0, State) ->
+    Method = cowboy_req:method(Req0),
+    Path = cowboy_req:path(Req0),
+    case {Method, Path} of
+        {<<"GET">>, <<"/">>} ->
+            handle_root(Req0, State);
+        {<<"GET">>, <<"/api/v1/items">>} ->
+            handle_get_items(Req0, State);
+        {<<"GET">>, <<"/api/v1/items/", ItemName/binary>>} ->
+            handle_get_item(Req0, State, ItemName);
+        {<<"POST">>, <<"/api/v1/items">>} ->
+            handle_post_items(Req0, State);
+        {<<"PUT">>, <<"/api/v1/items/", ItemName/binary>>} ->
+            handle_put_item(Req0, State, ItemName);
+        {<<"PATCH">>, <<"/api/v1/items/", ItemName/binary>>} ->
+            handle_patch_item_price(Req0, State, ItemName);
+        {<<"DELETE">>, <<"/api/v1/items/", ItemName/binary>>} ->
+            handle_delete_item(Req0, State, ItemName);
+        _ ->
+            handle_method_not_allowed(Req0, State)
+    end.
+
+handle_root(Req0, State) ->
+    Req = cowboy_req:reply(200,
+        #{<<"content-type">> => <<"text/plain">>},
+        <<"Welcome to the API Cowboy!">>,
+        Req0),
+    {ok, Req, State}.
+
+handle_get_items(Req0, State) ->
+    {ok, Client} = connect_to_redis(),
+    {ok, Keys} = eredis:q(Client, ["KEYS", "*"]),
```

Comment:
It's preferable to define a specific module to handle the DB operations and handle all the reds related logic like the keys. It's not a good definition to have a handler or a controller handling logic.